packages/good-design/src/advanced/customswitch/CustomSwitch.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useCallback } from 'react'
+import { Pressable, Icon, View, Box } from 'native-base'
+import { first } from 'lodash'
+
+import SelectBox from './SelectBox'
+// import SwitchIcon from ' ../../assets/svg/arrow-swap.svg'
+
+const SwitchIcon = () => (
+  <g strokeLinecap="round" strokeLinejoin="round" strokeWidth="1.5">
+    <path d="m5 10.0909h14l-5.8396-5.0909" stroke="#B0B4BB"/>
+    <path d="m19 13.9091h-14l5.8396 5.0909" stroke="#B0B4BB"/>
+  </g>
+)
+
+const SelectListItem = (
+  {
+    chain,
+    onPress,
+    isListItem,
+    isListOpen,
+    isLeft,
+    listNumber,
+  }: {
+    chain: string,
+    onPress: ((isLeft: boolean) => void) | ((isLeft: boolean, listNumber: number) => void)
+    isListItem: boolean,
+    isListOpen: boolean,
+    isLeft: boolean,
+    listNumber: number
+    
+  }) => {
+  const type = isListItem ? "list" : "button"
+  const onItemPress = useCallback(() => onPress(isLeft, listNumber), [chain, onPress]);
+  return (
+    <SelectBox variant={type} text={chain} press={onItemPress} isListItem={isListItem} isListOpen={isListOpen} />
+  )
+}
+
+//todo: add icon list (optional)
+export const CustomSwitch = ({list, switchListCb}:{list:string[], switchListCb: () => void}) => {
+  const [showListLeft, setShowListLeft] = useState<any>(false)
+  const [showListRight, setShowListRight] = useState<any>(false)
+
+  const [sourceList, setSourceList] = useState<string[]>(list);
+  const [targetList, setTargetList] = useState<string[]>(() => list.slice().reverse());
+
+  const toggleList = useCallback((left: boolean) => {
+    const side = {
+      show: left ? showListLeft : showListRight,
+      dispatch: left ? setShowListLeft : setShowListRight
+    }
+    side.dispatch(!side.show)
+  }, [showListLeft, showListRight])
+
+  const switchSelect = () => {
+    setSourceList(targetList);
+    setTargetList(sourceList);
+    switchListCb();
+  }
+
+  const selectFromList = (isLeft: boolean, listNumber: number) => {
+    const sides = [sourceList, targetList]
+    const selectedSide = isLeft ? sides.splice(0, 1)[0] : sides.splice(1,1)[0]
+    const altSide = sides[0]
+    const selected = selectedSide.splice(listNumber, listNumber)[0]
```

Comment:
```suggestion
    const listNumber = selectedSide.indexOf(chain)
    const selected = selectedSide.splice(listNumber, listNumber)[0]
```

Reply:
here we're getting index **at the moment** of item was checked/unchecked
so it won't break if items will be changed after first render

Reply:
Got it, changed